src/crystal/system/event_loop.cr

```diff
@@ -7,9 +7,30 @@ abstract class Crystal::EventLoop
     Crystal::Scheduler.event_loop
   end
 
-  # Runs the event loop.
+  # Runs the loop.
+  #
+  # Set `blocking` to false to return immediately if there are no activable
+  # events. Set it to true to wait for activable events, which will block the
+  # current thread until then.
+  #
+  # Returns `true` on normal returns (e.g. has activated events, has pending
+  # events but blocking was false) and `false` when there are no registered
+  # events.
+  abstract def run(blocking : Bool) : Bool
+
+  # Runs the event loop. Blocks until at least one event is activated then
+  # returns. May return immediately if there are activable events.
   abstract def run_once : Nil
```

Comment:
**question:** Is there a semantic difference between `run_once` and `run(blocking: true)`?
The existing implementations in this PR are identical. Is there any reason to think that future implementations would do anything different?

Reply:
No, we can replace it with `run(blocking: true)`.